src/helpers.js

```diff
@@ -330,7 +369,7 @@ const resolveRelativeLink = (url, courseData) => {
         page = parts.slice(parts.length - 1, parts.length)[0]
       }
       // build the base of the Hugo url
-      const newUrlBase = path.join("courses", courseId, "sections", ...sections)
+      const basePieces = ["/sections", ...sections]
```

Comment:
I wasn't exactly sure where to put this comment, as it has a bit wider of an implication than just resolving relative urls.  I tested this PR all the way through by generating markdown for all courses and then running it through the production build to make sure everything renders properly.  It seems that absolute URLs rendered in markdown do not respect the `baseUrl` argument / configuration value.  For example, if I use the starter to render all course sites using this command:

`./build.sh -o ~/Code/ocw-www/site/public -c ~/Code/ocw-to-hugo/private/test -b http://localhost:3000/courses`

...where `~/Code/ocw-to-hugo/private/test` is the markdown output from this PR, then I browse to the course `18-06sc-linear-algebra-fall-2011`, then to the `ax-b-and-the-four-subspaces/exam-1/` section, none of the links work because they do not respect `baseUrl` when rendered as part of `ocw-www`.  For example:

`http://localhost:3000/courses/18-06sc-linear-algebra-fall-2011/sections/ax-b-and-the-four-subspaces/exam-1/`

Then click the link for Exams (PDF) and you are brought to:

`http://localhost:3000/courses/18-06sc-linear-algebra-fall-2011/sections/ax-b-and-the-four-subspaces/exam-1/MIT18_06SCF11_ex1`

So, it seems we are not totally free of the need for some kind of special shortcode to help out here.  I think we might be able to approach it differently though, and simply create a `baseUrl` shortcode as these people have done:

https://discourse.gohugo.io/t/can-i-reference-baseurl-from-content/2686/2

As long as we prefix all absolute URLs in markdown with this, it should render the baseUrl argument before the URL, resulting in a fully qualified URL in the HTML.  I haven't tested this myself, but it could be another approach as opposed to bringing back `getpage`.

Reply:
What do you think about adding a `baseUrl` argument to ocw-to-hugo and having that construct prefixes for these URLs? This PR did that originally but I removed it because I thought it was unnecessary

Reply:
That's a possible solution, but seems like more work than just prefixing them with something like `{{< baseurl >}}` and adding a shortcode in `ocw-course-hugo-theme` that basically just has the content `{{ .Site.BaseURL }}`.  That way, when a site is generated using this content it will respect the `baseUrl` sent into the `hugo` command and you won't have to regenerate markdown to use a different one.

Reply:
That's a good point. I'll see about updating the PR for that